### ISSUE TITLE
Hiragana-only words do not show up in the dictionary

### ISSUE THREAD

**Issue by SpongebobSquamirez** [0xbad1d3a5/Kaku#24]
ぼうし(帽子)、けわしい(険しい)、うわさ(噂), and conjugated hiragana words can't be selected/aren't detected in the results window. I'm guessing if there's a kanji form, even if the kana form is common enough that it's written in kana in books for kids (as opposed to written in 漢字+振りがな), it won't be found in the dictionary.

**Comment by SpongebobSquamirez** [comment]
Just wanted to say that this is still a big issue for me. Seems like it never works. Was it supposed to work sometimes? I guess if a word has any associated kanzi, it just won't work. Also seems to affect different ways of writing the same word, eg compound verbals like いい表す (言い表す).

**Comment by 0xbad1d3a5** [comment]
I been pretty busy IRL the past month so haven't really had time to work on Kaku, unfortunately.

Hiragana lookups currently only work if the entry itself is in the dictionary character-for-character, which is often not the case. Supporting this requires rewriting the dictionary lookup logic since the current implementation is a pretty dumb one.

**Comment by SpongebobSquamirez** [comment]
Okay, thanks for letting me know. Just want to say that after consistent use of the app for months now, this is by far the issue that ends up costing me the most time, since it seems like half the time I look something up, this can't find the entry. If you find some time to work on the app in the near future, I think this is the most pressing issue.

**Comment by SpongebobSquamirez** [comment]
I know you're probably dead but since I still use this app all the time, and this is still the only thing that annoys the hell out of me about this app, and there don't seem to be any alternatives, I'm just gonna bump this in the off chance that you see this at some point and remember how much potential this thing has.

**Comment by p734** [comment]
I agree, everything else is not a big deal. Is there any way we can get more coverage to this app and someone can add it in maybe.

**Comment by dtan3847** [comment]
I ported over [Nazeka's](https://github.com/wareya/nazeka) search logic: https://github.com/dtan3847/Kaku
These examples work:
ぼうし(帽子)、けわしい(険しい)、うわさ(噂)
This example does not (it doesn't work in Nazeka):
いい表す

There are still problems (for example, right now all of the entries are loaded into memory) and maybe I'll have time to improve it later.

**Comment by 0xbad1d3a5** [comment]
Oh wow, you're a hero @dtan3847 haha.

**Comment by Artikash** [comment]
Hey all, I made a new database that basically fixes this by adding dictionary entries for kana readings. You can get it from my PR (above), then download the FDroid APK, open it with 7zip, replace the database (in assets folder), then re-sign and install the APK to use it. Or just download this APK [here](https://drive.google.com/file/d/137VeZc9nqrC_jeCEr2ETeLgY4_DY-sha/view?usp=sharing) if you trust me =)
